src/Components/Components/src/Routing/QueryParameterValueSupplier.cs

```diff
@@ -0,0 +1,186 @@
+                    {
+                        var decodedValue = suppliedPair.DecodeValue();
+
+                        foreach (ref var destination in CollectionsMarshal.AsSpan(destinations))
+                        {
+                            if (destination.IsArray)
+                            {
+                                valuesByDestination[destination.Index].Add(decodedValue);
+                            }
+                            else
+                            {
+                                valuesByDestination[destination.Index].SetSingle(decodedValue);
+                            }
+                        }
+                    }
+                }
+
+                // Finally, emit the parameter attributes by parsing all the string segments and building arrays
+                for (var destinationIndex = 0; destinationIndex < _destinations.Length; destinationIndex++)
+                {
+                    ref var destination = ref _destinations[destinationIndex];
+                    ref var values = ref valuesByDestination[destination.Index];
+
+                    var parsedValue = destination.IsArray
+                        ? destination.Parser.ParseMultiple(values, destination.ComponentParameterName)
+                        : values.Count == 0
+                            ? default
+                            : destination.Parser.Parse(values[0].Span, destination.ComponentParameterName);
+
+                    builder.AddAttribute(0, destination.ComponentParameterName, parsedValue);
+                }
+            }
+            finally
+            {
+                ArrayPool<StringSegmentAccumulator>.Shared.Return(valuesByDestination, true);
+            }
+        }
+
+        private static SortedDictionary<ReadOnlyMemory<char>, List<QueryParameterDestination>>? BuildQueryParameterMappings(Type componentType, out int destinationsCount)
+        {
+            var candidateProperties = MemberAssignment.GetPropertiesIncludingInherited(componentType, ComponentProperties.BindablePropertyFlags);
+            SortedDictionary<ReadOnlyMemory<char>, List<QueryParameterDestination>>? result = null;
+            destinationsCount = 0;
+
+            foreach (var propertyInfo in candidateProperties)
+            {
+                if (!propertyInfo.IsDefined(typeof(ParameterAttribute)))
+                {
+                    continue;
+                }
+
+                var fromQueryAttribute = propertyInfo.GetCustomAttribute<SupplyParameterFromQueryAttribute>();
+                if (fromQueryAttribute is not null)
+                {
+                    // Found a parameter that's assignable from querystring
+                    var componentParameterName = propertyInfo.Name;
+                    var queryParameterName = (string.IsNullOrEmpty(fromQueryAttribute.Name)
+                        ? componentParameterName
+                        : fromQueryAttribute.Name).AsMemory();
```

Comment:
I think if you drop the `AsMemory` here, it would do the conversion for you at the callsite (which should be largely free.

Reply:
Doesn't seem to be the case. There's an implicit conversion from `string` to `ReadOnlySpan<char>` but not to `ReadOnlyMemory<char>`.